### PR TITLE
Add ability to set up additional global pull credentials

### PR DESCRIPTION
**Pull request by zreigz**
You can create CRD `RegistryCredential` object with the new Docker registry credentials:
```
apiVersion: "platform.plural.sh/v1alpha1"
kind: RegistryCredential
metadata:
  name: demo-app-cred
  namespace: demo-app
spec:
  username: test
  email: demo@plural.sh
  server: dkr.plural.sh
  password: # reference to the secret with the password
    name: demo-app-secret
    key: password

```

Password secret:
```
apiVersion: v1
kind: Secret
metadata:
  name: demo-app-secret
  namespace: demo-app
data:
  password: cGFzc3dvcmQ=

```

 The controller creates a Secret with the `kubernetes.io/dockerconfigjson` type to authenticate with a container registry to pull a private image. 

The `secretsync_controller` copies credential Secret across namespaces.
The `serviceaccount_controller` updates ImagePullSecrets for the new entry.

You can find all example files in this directory: `test/registrycred`

### CODE REVIEW

File: controllers/registrycredential_controller.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	if err := r.createRegistryCredentialSecret(ctx, credentials); err != nil {
+		log.Error(err, "failed to create registry credential secret")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *RegistryCredentialsReconciler) createRegistryCredentialSecret(ctx context.Context, credentials platformv1alpha1.RegistryCredential) error {
+	if err := r.Get(ctx, client.ObjectKey{Namespace: credentials.Namespace, Name: credentials.GetPasswordSecretName()}, &corev1.Secret{}); err != nil {
```
Comment (michaeljguarino) [inline controllers/registrycredential_controller.go:90]: shouldn't we update the secret if it already exists?

**Review by michaeljguarino — COMMENTED** [review]
This mostly looks nice, one thing we should definitely do is bump the version so we don't overwrite the existing plural operator dkr image.  This should just be a matter of updating the VERSION file at the repo root

File: controllers/registrycredential_controller.go (hunk 1 above)
Comment (zreigz) [reply controllers/registrycredential_controller.go:90]: yes, good point I will add it

**Comment by zreigz** [comment]
@michaeljguarino I've added support for update (also password secret update) PTAL

File: controllers/registrycredential_controller.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      credentials.GetPasswordSecretName(),
+			Namespace: credentials.Namespace,
+		},
+	}
+	return r.Delete(ctx, secret)
+}
+
+// requestFromSecret returns a reconcile.Request for the credential registry if the secret is a password reference.
+func requestFromSecret(c client.Client) handler.EventHandler {
```
Comment (michaeljguarino) [inline controllers/registrycredential_controller.go:185]: can't we synthesize this with an owner reference or does that not necessarily work?  I'm fine with this approach, just curious.

Comment (zreigz) [reply controllers/registrycredential_controller.go:185]: the owner reference will work during deletion not necessarily with an update.

Comment (michaeljguarino) [reply controllers/registrycredential_controller.go:185]: gotcha good to know

**Review by michaeljguarino — APPROVED** [review]
So the release process for this is going to be a little different.  
* We need to bump the version for it (I believe there's a version file here).  That should be done before merging.  
* We'll also need to import the crds into `plural-artifacts` and wire up the new image version there as well 
* we'll also need to make sure the rbac for the deployment is updated too.

**Pull request merged**